dotnet/Cucumber.Messages/Cucumber.Messages.csproj

```diff
@@ -1,7 +1,7 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
+	
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.0</TargetFrameworks>
-    <LangVersion>10.0</LangVersion>
+    <TargetFrameworks>net8.0;netstandard2.0</TargetFrameworks>
```

Comment:
I have gone through this a few times, but I still not fully sure what benefit do we have with explicitly including .NET 8.0 as a target framework? My view on this so far was that this would be only useful (in a library!) if we would need any `#if` pragma in the code for the different targets.
@Romfos Could you please detail your intention a bit more in detail about the explicitly listed target?

Reply:
Yes, for example. We have method GetPackageVersion

![image](https://github.com/user-attachments/assets/ecbe73d9-8169-433a-a259-431cc338447f)

We can make a build for .NET Standard 2.0 tfm and .NET 8 tfm

![image](https://github.com/user-attachments/assets/6fec5304-9320-4f58-b3d4-25b7a565a1f2)
As you can see in one case we have an error, in second - we don't

Why this is working like that:
.NET 8 BCL and libraries have nullable reference type annotations. .NET Standard 2.0 is old and don't have it. By this reason .NET 8 version knows that GetCustomAttribute could return null, .NET Standard version - dont't know about it.
By this reason in .NET 8 version have additional warning about it

Why make sense to have a .NET 8 tfm in a list, event if you don't have a `#if` directives:
1) nullable reference types that lead to additional warnings (if Nullable enabled)
2) In general .NET team implemented bunch of new compiler warnings that help write more reliable code. Part of them could be backported to .NET Standard\.NET Framework. Some part - not
3) Compiler could generate more optimal IL code for modern .NET
4) New language features like `init` keyword, `requred` keyword, init records, e.t.c.

Reply:
Thx! I see, and let's keep it as it is then, but essentially some of these features are only about compiling our own code (so we could make a build for `-f net8.0` just to get the warnings, but do not include this into the package) or something (like `init`) that we anyway cannot use is we need to compile for .NET Standard as well. So the only one that is really applicable is the more optimal IL, but on the other side, the compilation time and the package size is doubled. Obviously for such small package all this does not really matter, it was more about my curiosity. Let's keep the double target. 😀